pkg/utils/kubernetes/client/client.go

```diff
@@ -277,11 +285,58 @@ func (o *cleaner) doClean(ctx context.Context, c client.Client, obj runtime.Obje
 	return nil
 }
 
+func hasFinalizers(obj runtime.Object) (bool, error) {
+	namespace, ok := obj.(*corev1.Namespace)
+	if !ok {
+		acc, err := meta.Accessor(obj)
+		if err != nil {
+			return false, err
+		}
+
+		return len(acc.GetFinalizers()) > 0, nil
+	}
+
+	return len(namespace.Finalizers)+len(namespace.Spec.Finalizers) > 0, nil
+}
+
+func doFinalize(ctx context.Context, k8sInterface kubernetes.Interface, obj runtime.Object) error {
+	_, ok := obj.(*corev1.Namespace)
+	if !ok {
+		return finalizeObject(ctx, k8sInterface, obj)
+	}
+
+	return finalizeNamespace(ctx, k8sInterface, obj)
+}
+
+func finalizeObject(ctx context.Context, k8sInterface kubernetes.Interface, obj runtime.Object) error {
+	acc, err := meta.Accessor(obj)
+	if err != nil {
+		return err
+	}
+
+	withFinalizers := obj.DeepCopyObject()
+	acc.SetFinalizers(nil)
+	return k8sInterface.Client().Patch(ctx, obj, client.MergeFrom(withFinalizers))
+}
+
+func finalizeNamespace(ctx context.Context, k8sInterface kubernetes.Interface, obj runtime.Object) error {
+	namespace, ok := obj.(*corev1.Namespace)
```

Comment:
For a namespace, you still need to also remove the regular finalizers.

Reply:
Setting both `.meta.finalizers` and `.spec.finalizers` and `CoreV1().Namespaces().Finalize(namespace)` does the trick.